File: js/structural/adapter.js
class OldLogger {
    write_log(text) {
        console.log(`old log: ${text}`)
    }
}

class LoggerAdapter {
    constructor(oldLogger) {
        this.oldLogger = oldLogger
    }

    log(oldLogger, text) {
        this.oldLogger.write_log(text)
    }
}

const client = (logger, text) => {
    logger.log(text)
}

oldLogger = new OldLogger()
adapterLogger = new LoggerAdapter(oldLogger)


client(adapterLogger, 'hello adapter pattern')

File: python/structural/adapter.py
class OldLogger:
    @staticmethod
    def write_log(text):
        print(f'old log: {text}')


# Adapter
class LoggerAdapter:
    def __init__(self, old_logger: OldLogger):
        self.old_logger = old_logger

    def log(self, msg):
        self.old_logger.write_log(msg)


def client(logger: LoggerAdapter, text):
    logger.log(text)


old_logger = OldLogger()
logger_adapter = LoggerAdapter(old_logger)

client(logger_adapter, 'hello adapter pattern')
